hail/src/main/scala/is/hail/expr/types/physical/PCanonicalString.scala

```diff
@@ -16,14 +16,12 @@ abstract class PCanonicalString(val required: Boolean) extends PString {
   lazy val binaryFundamentalType: PBinary = PBinary(required)
 
   def copyFromType(mb: MethodBuilder, region: Code[Region], srcPType: PType, srcAddress: Code[Long], forceDeep: Boolean): Code[Long] = {
-    assert(srcPType isOfType this)
```

Comment:
not needed because PString's only argument is required, so this is really just a check on isInstanceOf.